Add testing REST endpoints guide (#59)
wso2/docs-devant: en/docs/testing/test-rest-endpoints-via-the-openapi-console.md

# Test REST Endpoints via the OpenAPI Console

Devant provides an integrated OpenAPI Console for testing REST endpoints of your deployed integrations. Since Devant secures REST APIs with OAuth 2.0 authentication, the console automatically generates test keys to facilitate API testing.

Follow these steps to test a REST endpoint using the OpenAPI Console:

1. Go to the [Devant Console](https://console.devant.dev/) and log in.
2. Select the project.
3. In the **Integrations** pane, click on the integration you want to test.
4. On the desired environment card, click **Test** to open the **OpenAPI Console**.
5. In the **OpenAPI Console** pane, choose the endpoint you want to test from the **Endpoint** list.

    !!! note

>         - The **Network Visibility** of an endpoint is configured in the `component.yaml` file. You can set it to `Public`, `Organization`, or `Project`.

Reviewer: Shall we refactor this to
```suggestion
        - The **Network Visibility** of an endpoint is configured in the Configurations Drawer, which appears when you click `Configure` in a specific environment on the Integration Overview page. You can set the visibility to `Public`, `Organization`, or `Project`.
```

Sorry, missed this earlier

# Test REST Endpoints via the OpenAPI Console

> Devant offers an integrated OpenAPI Console to test REST endpoints for the Service components you create and deploy. Since Devant secures REST APIs with OAuth 2.0 authentication, the OpenAPI Console generates test keys to help you test your APIs.

Reviewer: ```suggestion
Devant provides an integrated OpenAPI Console for testing REST endpoints of your deployed integrations. Since Devant secures REST APIs with OAuth 2.0 authentication, the console automatically generates test keys to facilitate API testing.```

> Devant offers an integrated OpenAPI Console to test REST endpoints for the Service components you create and deploy. Since Devant secures REST APIs with OAuth 2.0 authentication, the OpenAPI Console generates test keys to help you test your APIs.

Author: ```suggestion
Devant provides an integrated OpenAPI Console for testing REST endpoints of your deployed integrations. Since Devant secures REST APIs with OAuth 2.0 authentication, the console automatically generates test keys to facilitate API testing.
```

Follow these steps to test a REST endpoint using the OpenAPI Console:

1. Go to the [Devant Console](https://console.devant.dev/) and log in.
2. Select the project.

> 3. In the **Integrations** pane, click on the component you want to test.

Reviewer 2: ```suggestion
3. In the **Integrations** pane, click on the integration you want to test.
```
In devant we use `integration` instead of `component`

> 4. Click **Test** in the left navigation menu, then select **Console**. This will open the **OpenAPI Console** pane.

Author: Fixed via [4dcea89](https://github.com/wso2/docs-devant/pull/59/commits/4dcea89c85fbc5f21aa7769a876bdc0bb3e7ab60)

5. In the **OpenAPI Console** pane, select the desired environment from the drop-down menu.
6. Choose the endpoint you want to test from the **Endpoint** list.

    !!! note

>         - The **Network Visibility** of an endpoint is configured in the **component.yaml** file. You can set it to **Public**, **Organization**, or **Project**.

Reviewer 2: ```suggestion
        - The **Network Visibility** of an endpoint is configured in the `component.yaml` file. You can set it to **Public**, **Organization**, or **Project**.
```

Author: Fixed via [4dcea89](https://github.com/wso2/docs-devant/pull/59/commits/4dcea89c85fbc5f21aa7769a876bdc0bb3e7ab60)

>         - The **Network Visibility** of an endpoint is configured in the **component.yaml** file. You can set it to **Public**, **Organization**, or **Project**.

Reviewer 2: ```suggestion
        - The **Network Visibility** of an endpoint is configured in the **component.yaml** file. You can set it to `Public`, `Organization`, or `Project`.
```
Values should be as code. Update the other places as well

Author: Fixed via [4dcea89](https://github.com/wso2/docs-devant/pull/59/commits/4dcea89c85fbc5f21aa7769a876bdc0bb3e7ab60)

        - If the **Network Visibility** is set to **Public**, skip to step 8.

>         - If the **Network Visibility** is set to **Organization**, the endpoint is not publicly accessible. However, Devant allows you to generate a temporary test URL for testing purposes. This URL is publicly accessible (secured with OAuth 2.0) and remains active for 15 minutes. To generate the temporary URL, proceed to step 6.

Author: Fixed via [4dcea89](https://github.com/wso2/docs-devant/pull/59/commits/4dcea89c85fbc5f21aa7769a876bdc0bb3e7ab60)

7. Generate a temporary test URL (if applicable):  

>    If the **Network Visibility** is set to **Organization**, click **Generate URL** to create a temporary test URL. This URL will be active for 15 minutes. If the visibility is not set to **Organization**, skip this step.

Reviewer: Let's not reiterate the same info here. May be we can say the expiry duration here instead in the prev step

If Network Visibility is set to Organization, click Generate URL to create a temporary test URL, valid for 15 minutes. If not, skip this step.

Author: Fixed via [4dcea89](https://github.com/wso2/docs-devant/pull/59/commits/4dcea89c85fbc5f21aa7769a876bdc0bb3e7ab60)

    !!! note
        The temporary test URL is valid for **15 minutes only** and will expire afterward. You can manually deactivate it by clicking the **Deactivate URL** button.

    ![OpenAPI Console](../assets/img/testing/openapi-console-org.png){.cInlineImage-full}

8. Expand the resource you want to test.

> 9. Click the **Try it out** button to enable testing.

Reviewer 2: ```suggestion
9. Click **Try it out** to enable testing.
```

Reviewer 2: Follow `click <button name>` style.
